client/src/app/employee/employee-detail/state/employee-detail.store.ts

```diff
@@ -0,0 +1,13 @@
+import { Injectable } from '@angular/core';
+import { EntityState, Store, StoreConfig } from '@datorama/akita';
+import { createInitialState, Employee } from './employee-detail.model';
+
+export interface EmployeeDetailState extends EntityState<Employee> {}
+
+@Injectable({ providedIn: 'root' })
+@StoreConfig({ name: 'employee-detail' })
+export class EmployeeDetailStore extends Store<EmployeeDetailState> {
```

Comment:
Did you make these with the Angular schematics?

If your state extends `EntityState<T>` then you should be using `EntityStore` and `QueryEntity`.

Then you could create a custom `Observable<>` using the `selectActive()` method that selected the first active employee.

Reply:
I did make them with the Angular Schematics, not 100% sure i made the right decisions.

Reply:
Updated this `query`, `store`, `service`, and `model` to reflect the naming convention below since this `store` is currently retrieving an `Employee`, I can refactor naming again once we start utilizing the entire `EmployeeDetail` object graph.